mewarpx/mewarpx/setups_store/diode_setup.py

```diff
@@ -626,10 +621,25 @@ def init_resultsinfo(self):
         # )
         self.control = SimControl(max_steps=self.TOTAL_TIMESTEPS)
 
-    def init_warpx(self):
-        sim = picmi.Simulation(
+    def init_simulation(self, **kw):
+        warpx_collisions = kw.pop('warpx_collisions')
+        self.sim = picmi.Simulation(
                     solver = self.solver,
                     time_step_size = self.DT,
-                    max_steps = self.TOTAL_TIMESTEPS)
+                    max_steps = self.TOTAL_TIMESTEPS,
+                    warpx_collisions=warpx_collisions)
+        return self.sim
+
+    def init_warpx(self, **kw):
+        if self.sim is None:
+            self.sim = self.init_simulation(**kw)
 
-        mwxrun.init_run(simulation=sim)
\ No newline at end of file
+        mwxrun.init_run(simulation=self.sim)
+
+    def add_species_to_sim(self, species, particle_per_cell):
```

Comment:
This should be able to happen with input parameters like most things in here too, rather than called separately. I think in Warp it was part of the setup injectors area, but we can move stuff around.